Sources/Core/JavaIR.swift

```diff
@@ -135,22 +173,40 @@ public struct JavaIR {
             let implementsList = implements?.joined(separator: ", ") ?? ""
             let implementsStmt = implementsList == "" ? "" : "implements \(implementsList)"
 
-            var propertiesStrings = [String]()
-            for propertyBatch in properties {
-                for property in propertyBatch {
-                    propertiesStrings.append(property.render())
+            let extendsStmt = extends.map { " extends \($0) " } ?? ""
+
+            var toRender = annotations.map { "@\($0)" } + [
+                "\(modifiers.render()) class \(name)\(extendsStmt) \(implementsStmt) {",
+            ]
+
+            if !enums.isEmpty {
+                toRender.append("")
+                toRender.append(-->enums.flatMap { $0.render() })
+            }
+
+            if !properties.isEmpty {
+                toRender += properties.reduce([String]()) { result, propertyBatch in
+                    result + [""] + propertyBatch.map { property in
+                        -->[property.render()]
+                    }
                 }
-                propertiesStrings.append("") // Add an empty line between each batch
             }
 
-            return annotations.map { "@\($0)" } + [
-                "\(modifiers.render()) class \(name) \(implementsStmt) {",
-                -->enums.flatMap { $0.render() },
-                -->propertiesStrings,
-                -->methods.flatMap { $0.render() },
-                -->innerClasses.flatMap { $0.render() },
-                "}",
-            ]
+            if !methods.isEmpty {
+                toRender += (methods.reduce([String]()) { result, method in
+                    result + [""] + [-->method.render()]
+                })
+            }
+
+            if !innerClasses.isEmpty {
```

Comment:
It seems like the issue with a lot of the formatting here is just handling empty scopes? My only worry is that I feel like the code has becomes a bit less readable / maintainable after these changes.

Reply:
It was to handle a few things:
1. Avoid having an empty line if the enums/properties/methods/innerClasses is empty. 
2. Avoid having tabbed empty lines. I think I can solve this in a different way by tweaking the indent() string extension function.
3. Other little tweaks like adding an empty line in between each method.

I tried to simplify it a bit. Let me know if you still think it's too un-readable.

Reply:
>  by tweaking the indent() string extension function.
Ended up tweaking --> instead of indent()

Reply:
Thank you, this looks a lot better! Did you get a chance to run `make integration_test` after? I think this might alter the output of the other languages